hoprd/db/api/src/metadata.rs

```diff
@@ -0,0 +1,261 @@
+use async_trait::async_trait;
+use hopr_primitive_types::prelude::*;
+use sea_orm::{ColumnTrait, EntityTrait, QueryFilter};
+
+use crate::{db::HoprdDb, errors::Result};
+
+/// Defines DB API for accessing HOPR settings (mainly aliases for now)
+#[async_trait]
+pub trait HoprdDbMetadataOperations {
+    /// Retrieve the alias for a given PeerID
+    async fn resolve_alias(&self, alias: String) -> Result<Option<String>>;
+
+    /// Retrive all aliases
+    async fn get_aliases(&self) -> Result<Vec<Alias>>;
+
+    /// Create new key pair value in the db
+    async fn set_alias(&self, peer: String, alias: String) -> Result<()>;
+
+    /// Update aliases
+    async fn set_aliases(&self, aliases: Vec<Alias>) -> Result<()>;
+
+    /// Delete alias
+    async fn delete_alias(&self, alias: String) -> Result<()>;
+
+    /// Delete all aliases
+    async fn clear_aliases(&self) -> Result<()>;
+}
+
+#[async_trait]
+impl HoprdDbMetadataOperations for HoprdDb {
+    async fn resolve_alias(&self, alias: String) -> Result<Option<String>> {
+        let row = hoprd_db_entity::aliases::Entity::find()
+            .filter(hoprd_db_entity::aliases::Column::Alias.eq(alias))
+            .one(&self.metadata)
+            .await?;
+
+        Ok(row.map(|model| model.peer_id))
+    }
+
+    async fn get_aliases(&self) -> Result<Vec<Alias>> {
+        let rows = hoprd_db_entity::aliases::Entity::find().all(&self.metadata).await?;
+
+        let aliases = rows
+            .iter()
+            .map(|row| Alias {
+                peer_id: row.peer_id.clone(),
+                alias: row.alias.clone(),
+            })
+            .collect();
+
+        Ok(aliases)
+    }
+
+    async fn set_aliases(&self, aliases: Vec<Alias>) -> Result<()> {
```

Comment:
What is the semantics of this API? 
Should it insert always (= i.e. fail if some of the aliases exists?)
Should it upsert? (= insert or update existing ones)
Should it insert but ignore if some alias exists?

This needs to be explained in the method documentation on the trait.